stellarphot/settings/tests/test_custom_widgets.py

```diff
@@ -401,6 +405,151 @@ def test_revert_button_remains_enabled_with_invalid_value_and_actually_reverts(
         assert choose_or_make_new._item_widget.value == TEST_CAMERA_VALUES
 
+    def test_delete_button_click_displays_confirm_dialog(self, tmp_path):
+        # Clicking "Delete" should display a confirmation dialog
+
+        # Make an item to ensure an existing item is displayed when the widget is made
+        self.make_test_camera(tmp_path)
+
+        choose_or_make_new = ChooseOrMakeNew("camera", _testing_path=tmp_path)
+        choose_or_make_new._delete_button.click()
+
+        # Confirm dialog should be displayed
+        assert choose_or_make_new._confirm_edit_delete.layout.display != "none"
+        # edit/delete buttons should be hidden
+        assert choose_or_make_new._edit_delete_container.layout.display == "none"
+        # The confirm dialog should indicate you are deleting
+        assert "delete" in choose_or_make_new._confirm_edit_delete.message.lower()
+
+    @pytest.mark.parametrize("click_yes", [True, False])
+    def test_delete_actions_after_confirmation(self, tmp_path, click_yes):
+        # Clicking "Yes" in the confirmation dialog should delete the item
+        self.make_test_camera(tmp_path)
+
+        choose_or_make_new = ChooseOrMakeNew("camera", _testing_path=tmp_path)
+        choose_or_make_new._delete_button.click()
+
+        if click_yes:
+            # Simulate a click on the "Yes" button in the confirm dialog
+            choose_or_make_new._confirm_edit_delete._yes.click()
+
+            # The camera should be gone
+            saved = SavedSettings(_testing_path=tmp_path)
+            cameras = saved.get_items("camera")
+            assert len(cameras.as_dict) == 0
+        else:
+            # Simulate a click on the "No" button in the confirm dialog
+            choose_or_make_new._confirm_edit_delete._no.click()
+
+            # The camera should still be there
+            saved = SavedSettings(_testing_path=tmp_path)
+            cameras = saved.get_items("camera")
+            assert len(cameras.as_dict) == 1
+            assert list(cameras.as_dict.values())[0].model_dump() == TEST_CAMERA_VALUES
+
+        # Regardless of the choice, the confirm dialog should be hidden
+        assert choose_or_make_new._confirm_edit_delete.layout.display == "none"
+
+        # We should also no longer be deleting
+        assert not choose_or_make_new._deleting
+
+        # If the user clicked "No" the edit/delete buttons should be displayed because
+        # there is still an item to edit or delete.
+        if not click_yes:
+            assert choose_or_make_new._edit_delete_container.layout.display != "none"
+        # If the user clicked "Yes" the edit/delete buttons should be hidden because
+        # there is no item to edit or delete.
+        else:
+            assert choose_or_make_new._edit_delete_container.layout.display == "none"
+
+    def test_correct_item_selected_after_delete_and_yes(self, tmp_path):
+        # The correct item should be selected after an item is deleted and the user
```

Comment:
Again, logical test.  And it makes sense to check although I wouldn't have thought of it.  I imagine the GUI interface is leading to all sorts of potential tests of things you wouldn't have originallyy thought of.